### PR TITLE
Setup docker and test with Docker

### PR DESCRIPTION
**Pull request by Baschdl**
This PR adds a Dockerfile and a script to deploy our system with Docker.

### CODE REVIEW

**Comment by Baschdl** [comment]
@jexp In the moment we've some trouble using the [docker image](https://github.com/neo4j/docker-neo4j/blob/master/src/3.3/Dockerfile) and [entrypoint.sh](https://github.com/neo4j/docker-neo4j/blob/master/src/3.3/docker-entrypoint.sh) provided by neo4j. We want to automatically build our changes to neo4j-graph-algorithms in a [docker image on top of neo4j's docker image](https://github.com/KDD-OpenSource/32de-python/blob/configure-docker/Dockerfile-neo4j-graph-algorithms).
What is the intended way to expose the `plugins` folder of neo4j to the outside so we are able to copy our built graph algorithm jars to it? We tried [to define it as a volume](https://github.com/KDD-OpenSource/32de-python/blob/566882b0034130f07714d3a94066118ada67f065/docker-deployment.sh#L15) and [tried the parameter of the entrypoint.sh](https://github.com/KDD-OpenSource/32de-python/blob/566882b0034130f07714d3a94066118ada67f065/docker-deployment.sh#L20).

**Comment by jexp** [comment]
you can just expose it as a volume http://neo4j.com/docs/operations-manual/current/installation/docker/#docker-overview

**Comment by Baschdl** [comment]
Thanks. That is working now. @jexp What is the intended way to start a image built on neo4j's image? The [same lines](https://github.com/KDD-OpenSource/32de-python/blob/779ccd409ca4b0bc5f9766ad556709419be1cae8/Dockerfile-neo4j-graph-algorithms#L9) as in the neo4j repo don't work. The `/docker-entrypoint.sh` file should be in the right place but I'm not completely sure.

**Comment by Baschdl** [comment]
The neo4j problem still persists but the Dockerfile is moved to KDD-OpenSource/neo4j-graph-algorithms:alpha-dev. So it's mergeable from the python part.

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# List of packages which are needed for development and deployment
 numpy==1.14.0
 sklearn
 typing
```
Comment (MkuuWaUjinga) [inline requirements.txt]: I think _typing_ is already included in the standard library

Comment (Baschdl) [reply requirements.txt]: But only for newer python versions, old ones need this extra package.

**Pull request merged**